lib/biker.rb

```diff
@@ -1,3 +1,25 @@
+require './lib/ride.rb'
+
 class Biker
+    attr_reader :name,
+                :max_distance,
+                :rides,
+                :acceptable_terrain
+
+    def initialize(name, max_distance)
+        @name = name
+        @max_distance = max_distance
+        @rides = {}
+        @acceptable_terrain = []
+    end
+
+    def learn_terrain!(terrain)
+        if !@acceptable_terrain.include?(terrain)
+            @acceptable_terrain << terrain
+        end
+    end  
 
+    def log_ride(ride, distance)
+        @rides[ride] = @rides.fetch(ride, []) << distance
```

Comment:
This is a very nice option for getting a default [] when the key doesn't exist. Good use of `fetch`

Reply:
The only piece missing from this function is the condition to not let bikers log rides if they haven't learned the terrain or have a max_distance that's lower than the ride distance. Consider adding another if statement OR using a guard clause with these conditions